### PR TITLE
everywhere: reindent `.overlay` files with tabs

### PR DESCRIPTION
**Pull request by JordanYates**
`checkpatch.pl` requires that dts sources are indented with tabs, fix all the spaces that slipped in while checkpatch wasn't watching (#74570).

### CODE REVIEW

File: boards/shields/esp_8266/boards/numaker_pfm_m467.overlay [hunk 1]
```diff
@@ -5,14 +5,14 @@
  */
 
 &pinctrl {
-    uart2_esp8266: uart2_esp8266 {
+	uart2_esp8266: uart2_esp8266 {
 	group0 {
```
Comment (jeppenodgaard) [inline boards/shields/esp_8266/boards/numaker_pfm_m467.overlay:9]: Needs indentation.

File: samples/drivers/spi_flash/boards/stm32h573i_dk.overlay [hunk 2]
```diff
@@ -5,9 +5,9 @@
  */
 
 &xspi1 {
-       /* request 57 for XSPI1 */
+	/* request 57 for XSPI1 */
 	dmas = <&gpdma1 4 57 STM32_DMA_PERIPH_TX
-		&gpdma1 5 57 STM32_DMA_PERIPH_RX>;
+			&gpdma1 5 57 STM32_DMA_PERIPH_RX>;
```
Comment (jeppenodgaard) [inline samples/drivers/spi_flash/boards/stm32h573i_dk.overlay:10]: Is tabs and then a few spaces allowed to align entries? If so, this and other similar changes should probably remain unchanged.

**Review by jeppenodgaard — COMMENTED** [review]
Nice to get this fixed!

File: samples/drivers/spi_flash/boards/stm32h573i_dk.overlay (hunk 2 above)
Comment (JordanYates) [reply samples/drivers/spi_flash/boards/stm32h573i_dk.overlay:10]: I was performing the alignment based on 4 character tabs, which I now see is not the default for dts source. I will do another parse over the files

**Review by jeppenodgaard — APPROVED** [review]

**Review by jeppenodgaard — APPROVED** [review]

**Review by nashif — DISMISSED** [review]

**Review by aasinclair — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]

**Review by ubieda — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
Please rebase on `main` and resolve any merge conflicts.

**Comment by JordanYates** [comment]
> Please rebase on `main` and resolve any merge conflicts.

done

**Review by kartben — DISMISSED** [review]

**Review by aasinclair — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Review by aescolar — DISMISSED** [review]

**Comment by aescolar** [comment]
@JordanYates Github demands a rebase for some reason..

**Comment by JordanYates** [comment]
> @JordanYates Github demands a rebase for some reason.. (and your original branch is protected so I can't push the rebase directly to it)

Odd indeed, no conflicts appeared while rebasing...
Done

**Review by aescolar — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by simhein — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]
thanks!

**Pull request merged**